tests/src/main/java/org/hibernate/beanvalidation/tck/tests/validation/graphnavigation/containerelement/NestedCascadingOnContainerElementsTest.java

```diff
@@ -82,6 +86,41 @@ public void testNestedValid() {
 
 	@Test
 	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_REQUIREMENTS_GRAPHVALIDATION, id = "l")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_REQUIREMENTS_GRAPHVALIDATION, id = "m")
+	public void testNestedValidWithTwoInnerExtractions() {
+		Validator validator = getValidator();
+
+		Set<ConstraintViolation<AddressBook>> constraintViolations = validator.validate( AddressBook.valid() );
+
+		assertNumberOfViolations( constraintViolations, 0 );
+
+		constraintViolations = validator.validate( AddressBook.invalid() );
+
+		assertThat( constraintViolations ).containsOnlyViolations(
+				violationOf( NotBlank.class )
+						.withPropertyPath( pathWith()
+								.property( "addressesPerTypePerPerson" )
+								.containerElement( "<map value>", true, "Firstname Lastname", null, Map.class, 1 )
```

Comment:
It is :).

It's a nested Map with 2 levels: it's the value on the first level and the key for the second. See the line below, the `typeArgumentIndex` is 0.

(btw, unrelated, but I force pushed a minor cleanup as there was `@Valid` annotations wandering around in the Map initialization code)